gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/tasks/AbstractUnpackDefaultComposeApplicationResourcesTask.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.desktop.tasks
+
+import org.gradle.api.file.Directory
+import org.gradle.api.file.FileSystemLocation
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.jetbrains.compose.ComposeBuildConfig
+import org.jetbrains.compose.desktop.application.internal.ioFile
+
+abstract class AbstractUnpackDefaultComposeApplicationResourcesTask : AbstractComposeDesktopTask() {
+    internal class DefaultResourcesProvider(resourcesRootDir: Provider<Directory>) {
+        val macIcon: Provider<RegularFile> = resourcesRootDir.map { it.file("default-icon-mac.icns") }
```

Comment:
Are those names somehow documented?

Reply:
No, and I don't think either of them is a part of public contract. After all, an icon is copied to a `<APP_NAME>.app` as `<APP_NAME>.icns`